File: src/main/java/maifetch/Ansi.java
package maifetch;

public final class Ansi {
    private Ansi() {
    }

    public static String foreground(String text, int red, int green, int blue) {
        return "\u001B[38;2;" + red + ";" + green + ";" + blue + "m" + text + "\u001B[0m";
    }

    public static String foregroundBackground(
            String text,
            int foregroundRed,
            int foregroundGreen,
            int foregroundBlue,
            int backgroundRed,
            int backgroundGreen,
            int backgroundBlue
    ) {
        return "\u001B[38;2;" + foregroundRed + ";" + foregroundGreen + ";" + foregroundBlue
                + ";48;2;" + backgroundRed + ";" + backgroundGreen + ";" + backgroundBlue
                + "m" + text + "\u001B[0m";
    }
}

File: src/main/kotlin/maifetch/Ansi.kt
package maifetch

object Ansi {
    fun fg(text: String, r: Int, g: Int, b: Int): String = "\u001b[38;2;$r;$g;${b}m$text\u001b[0m"

    fun color(text: String, fgR: Int, fgG: Int, fgB: Int, bgR: Int, bgG: Int, bgB: Int): String =
        "\u001b[38;2;$fgR;$fgG;${fgB}m\u001b[48;2;$bgR;$bgG;${bgB}m$text\u001b[0m"
}
